orca/topology/infra/k8s/matcher.py

```diff
@@ -218,6 +218,23 @@ def are_linked(self, obj, namespace):
         return namespace.properties.name == obj.properties.namespace
 
 
+class IngressToServiceMatcher(Matcher):
+
+    """Generic matcher for links between Ingress and Service entities."""
+
+    def are_linked(self, ingress, service):
+        matched_namespace = match_namespace(ingress, service)
+        matched_service = self._match_service(ingress, service)
+        return matched_namespace and matched_service
+
+    def _match_service(self, ingress, service):
```

Comment:
Since we're matching rules here, I suggest renaming the method to `_match_rules`.